packages/cache/src/CacheStorage.ts

```diff
@@ -66,8 +85,10 @@ export abstract class CacheStorage implements ICacheStorage {
 
     this.logger.setHit(result);
 
-    // Save hash of files if not already memoized
-    fetchHashesFor(this.cwd);
+    if (excludeUnchanged) {
```

Comment:
I see we used to `fetchHasesFor` all the time before. Do we not need it anymore in cases where we're not using the `BACKFILL_EXCLUDE_UNCHANGED` env var?

Reply:
No, it was a mistake in my previous PR :(

Reply:
I see that there's logic in `put` related to only storing changed files. Has this been in prod for a while now?

Reply:
yes, we pushed it to main to be able to test the change in the CI of midgard but it did not work because of a bug (that is now fixed). The feature is not turned on anywhere at the moment.

Reply:
I just realized this is an iteration of an earlier version of this same functionality, so this comment is not relevant.